Source/SwiftLintFramework/Rules/UntypedErrorRule.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  UntypedErrorRule.swift
+//  SwiftLint
+//
+//  Created by Daniel.Metzing on 17/02/18.
+//  Copyright © 2018 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct UntypedErrorRule: ASTRule, OptInRule, ConfigurationProviderRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    private static let regularExpression = regex(
+        "catch" + // The catch keyword
+        "(?:"   + // Start of the first non-capturing group
+        "\\s+"  + // At least one any type of whitespace character
+        "(?:"   + // Start of the alternative non-capturing group
+        "let"   + // `let` keyword
+        "|"     + // OR
+        "var"   + // `var` keyword
+        ")"     + // End of the alternative non-capturing group
+        "\\s+"  + // At least one any type of whitespace character
+        "\\w+"  + // At least one any type of word character
+        ")"     + // End of the first non-capturing group
+        "(?:"   + // Start of the second non-capturing group
+        "\\s*"  + // Zero or unlimited any whitespace character
+        ")"     + // End of the second non-capturing group
+        "\\{"     // Start scope character
+    )
+
+    public static let description = RuleDescription(
+        identifier: "untyped_error",
+        name: "Untyped error rule",
+        description: "Catch statements should not declare error variables without type casting",
+        kind: .lint,
+        nonTriggeringExamples: [
+            "do {\n try foo() \n} catch {\n}",
+            "do {\n try foo() \n} catch Error.invalidOperation {\n} catch {\n}",
+            "do {\n try foo() \n} catch let error as? MyError {\n} catch {\n}",
+            "do {\n try foo() \n} catch var error as? MyError {\n} catch {\n}"
+        ],
+        triggeringExamples: [
+            "do {\n try foo() \n} ↓catch var error {\n}",
+            "do {\n try foo() \n} ↓catch let error {\n}",
+            "do {\n try foo() \n} ↓catch let someError {\n}",
+            "do {\n try foo() \n} ↓catch var someError {\n}",
+            "do {\n try foo() \n} \n ↓catch let e {\n}"
+        ])
+
+    public func validate(file: File,
+                         kind: StatementKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+
+        guard kind == .brace,
+            let offset = dictionary.offset else {
+                return []
+        }
+
+        return file.lines
+            .lazy
```

Comment:
Does this really improve performance? I think `lines` will be calculated anyways.